services/user_service.py

```diff
@@ -30,4 +30,26 @@ def remove_user(self, email: str):
         user = self.user_repository.get_user_by_email(email)
         if user is None:
             raise ValueError("User does not exist.")
-        return self.user_repository.delete_user(email)
\ No newline at end of file
+        self.user_repository.reinitialize_mentees_mentor(email)
+        return self.user_repository.delete_user(email)
+    
+    def register_mentor(self, mentor_email: str, mentee_email: str):
+        mentor = self.user_repository.get_user_by_email(mentor_email)
+        if mentor is None:
+            raise ValueError("User does not exist !!!")
```

Comment:
A more descriptive message would be better. Something like `Cannot assign mentor with email <email> because user do not exist !!!`